src/proxy.cc

```diff
@@ -919,6 +924,44 @@ ncclResult_t ncclProxyProgressDestroy(struct ncclProxyState* proxyState) {
   return ncclSuccess;
 }
 
+void *rcclProxyMonitor(void *args) {
+  int interval = *(int*) args;
+  int elapsed = 0;
+
+  while (proxyMonitorInit) {
+    if (elapsed > interval) {
+      dumpProxyState(ncclLastProxyState);
+      elapsed = 0;
+    }
+    sleep(1);
+    elapsed += 1;
+  }
+
+  free(args);
+  return NULL;
+}
+
+static ncclResult_t rcclProxyMonitorCreate() {
+  if(rcclParamProxyMonitorInterval() <= 0) return ncclSuccess;
+  if(proxyMonitorInit == 1) return ncclSuccess;
+
+  int *interval = (int*) malloc(sizeof(int));
+
+  *interval = rcclParamProxyMonitorInterval();
```

Comment:
can you just read `rcclParamProxyMonitorInterval` as a local var inside `proxyMonitorThread`? I didn't see the need to pass it as arg